src/test/java/com/epam/tc/hw4/ex1/Exercise1Test.java

```diff
@@ -0,0 +1,94 @@
+package com.epam.tc.hw4.ex1;
+
+import static io.qameta.allure.Allure.step;
+
+import com.epam.tc.hw3.page.objects.composite.HomePage;
+import com.epam.tc.hw3.page.objects.composite.component.FrameComponent;
+import com.epam.tc.hw4.BaseExerciseTestHW4;
+import com.epam.tc.hw4.storynames.Tags;
+import com.epam.tc.hw4.utils.DataProviderForHW4;
+import io.qameta.allure.Description;
+import io.qameta.allure.Epic;
+import io.qameta.allure.Feature;
+import io.qameta.allure.Step;
+import io.qameta.allure.Story;
+import java.util.List;
+import org.assertj.core.api.SoftAssertions;
+import org.testng.annotations.Test;
+
+@Feature(Tags.FEATURE_EX1)
+public class Exercise1Test extends BaseExerciseTestHW4 {
+
+    @Test(
+        dataProviderClass = DataProviderForHW4.class,
+        dataProvider = "EX1"
+    )
+    @Step
+    @Description("Exercise 1 test including testing login and general page elements")
+    @Story(Tags.STORY_EX1)
+    public void testLoginAndGeneralHomePageElements(String title, String[] expectedNavBarTitles,
+                                                    long iconsCount,
+                                                    String[] expectedTextsUnderIcons,
+                                                    String[] expectedMenuTitles) {
+        step("Open test site by URL");
+        driver.get(properties.getProperty("url"));
+
+        // set up the soft assertions
+        SoftAssertions softAssertions = new SoftAssertions();
+
+        // initialize homePage
+        HomePage homePage = new HomePage(driver);
+
+        step("Assert Browser title");
+        softAssertions.assertThat(homePage.getTitle()).isEqualTo(title);
+
+        step("Perform login");
+        homePage.topMenu().clickLoginForm();
+        homePage.topMenu().sendKeysName(properties.getProperty("username"));
+        homePage.topMenu().sendKeysPassword(properties.getProperty("password"));
+        homePage.topMenu().clickLoginButton();
```

Comment:
What do you think about moving login steps to HomePage or TopMenu page object?
Login can be used in many tests and usually it is a part of PO

Reply:
Created a method performLogin in TopMenu class.